### ISSUE TITLE
l10n_nl

### ISSUE THREAD

**Issue by lap-odoo** [odoo/odoo#18572]
Dear Odoo Partners,

At Odoo, a lot of effort is put into improving the accounting localizations for each country and also to maintain these accounting localizations up-to-date. To achieve that, we will need your help.

One github issue will be created per accounting localization (e.g. l10n_be for Belgium). We expect from you to make suggestions about what should be improved in the specific accounting localization. Do not hesitate to share files with detailed explanations.

As soon as changes will be implemented, you will be informed and will have the opportunity to test it and share your feedback.

This github issue will also be useful to keep you informed about all the ongoing improvements concerning a specific accounting localization.

**Comment by lap-odoo** [comment]
Dear Partners, 

We are already working on:
- Adding new tags on the CoA to get a more detailed P&L
- Improving the CoA
- The Intrastat Report 
- XAF Export

As soon as this will ready, I'll inform you so that you can test it. 

Thanks in advance for your help.

**Comment by Yenthe666** [comment]
@ploegvde I bet you can give feedback here!

**Comment by ploegvde** [comment]
Hi, we are already for over a month in contact with Odoo regarding to the Dutch Accounting. We will gladly test the new functionality.

**Comment by lap-odoo** [comment]
Hi, indeed, Erwin's team already helped a lot. As soon as something will be ready to be tested I'll inform you.

**Comment by ploegvde** [comment]
Hi, we are in need for a The Intrastat Report (ICP). This report is required by law in the Netherlands as an addon on the tax reporting. Can we have discussion on how the solution will be in V11, so that we maybe can create same solution in V10?

**Comment by smetl** [comment]
Hi! I'm facing to this code:

 https://github.com/OCA/l10n-netherlands/blob/10.0/l10n_nl_intrastat/models/l10n_nl_intrastat.py#L127-L131 

...concerning the Intrastat Report (ICP). Is it mandatory to exclude some invoice lines based on special boolean on the tax?

**Comment by ploegvde** [comment]
IMO it is not needed. The module you are referring to takes another approach. The report is made based on invoice lines. They new report should be based on account move lines. The total of label '3b. Leveringen naar/diensten in landen binnen de EU (omzet)' should be the same as the total of the ICP. The ICP list out the amount service and goods per customers. Per customer we show the name, country and Tax code.

So please do not look at the module from the NL OCA repo. It is migrated from V8 when account labels where not existing.

The current (Enterprise) Tax report and the ICP report should have the same basis and that is the account move line.

**Comment by ploegvde** [comment]
I also want to note that the ICP report is required by law and Odoo does not have a proper solution. So Odoo does not comply with Dutch regulations. In conjunction with that, when you deliver more that 1.200.000 to countries within the EU, also a CBS report is required by law. Also this report is not available in Odoo. Don't you think Odoo should comply and have these report also for at leas V10?

Ref Dutch Tax authorities: https://www.belastingdienst.nl/wps/wcm/connect/bldcontentnl/belastingdienst/zakelijk/btw/zakendoen_met_het_buitenland/goederen_en_diensten_naar_andere_eu_landen/administratie_bijhouden_van_goederen_en_diensten_naar_andere_eu-landen/

**Comment by Reddconnect-Maarten** [comment]
1.	Reconciliation of journal items in regards to “relations”:  
In the functionality of reconciliation of journal items, Odoo does not allow reconciliation of journal items which a linked to different relations (read Partner-Id’s). This constrain should be removed. Often, reconciliation on items from different relations is highly desired.     

2.	Bank reconciliation receivables and payables:
Odoo does not allow the include both receivable journal items as well as payable journal items to reconcile a bank transaction. This constrain should be removed.

3.	Sepa direct debit file: 
Create Sepa-direct-debit file based on the selected customer's invoices. At the moment of file creation, the payment-status of the customer's invoices which are included in the direct-debit file should be changed to “included in direct debit”. The sepa-batch-reference (is payment reference together with invoice number) should be registered on the invoices included.   

4.	Overdue report: 
A downloadable Excel report presenting a list of all Relations (res.partners) with all its outstanding (overdue) amounts (=balance) of each invoice-numbers as being accumulated for each overdue period: for example periods of 0-30 days, 31-60 days, 61-90 days, etc. At this moment, Odoo already delivers a kind-off report, called aged report receivables. However, this report is a dynamic report, what implies that you need the open first all the relations before you have the whole list you want. 

5.	Follow-up process of overdue invoices: 
In addition to the already available functionality of follow-up, we like to see improvement in the way follow-up messages are being selected to send out. We should be able to make a selection on the open-invoices list to include invoices in a follow-up batch-messages to send out together with the customer statements. The statement for a particular relation should include the selected invoice as well as the number of times invoices has been already being included (in the past) in the follow-up message. Therefore, it should we required to register the number of follow-ups being send-out on the invoice record. 

6.	Follow-up statement: 
we need to have the list of overdue invoices together with the original invoice amount and balance amount (= invoice amount deducted with registered payments). At the moment, the invoice amount is not being presented on the statement. 

7.	Send digital suppliers’ invoices (PDF) to Odoo. 
It should be good when users are able te send supplier-invoices directly to Odoo (similar to the expense-functionality). So when a suppliers’ invoices is being received by Odoo, it creates an concept-invoice with the PDF attached. Now the concept can be amended with the correct data and confirmed. 

8.	ICP rapport: 
For the selected period, the report represents all the sums for all amounts of EU-sale transactions accumulated for each VAT-number. The amount of sale transactions are being split in sum of amounts representing products and sum of amounts representing services.

**Comment by lap-odoo** [comment]
@ploegvde : Hi Erwin, 

Regarding the ICP & CBS, indeed we think these are important reports. However, we will try to improve the l10n_nl step by step. So we will first finish what we have started and merge it before starting to work on other requirements. I've taken note of the fact we still need to work on the ICP and CBS.

**Comment by lap-odoo** [comment]
@Reddconnect-Maarten 

Reconciliation of journal items in regards to “relations”:
In the functionality of reconciliation of journal items, Odoo does not allow reconciliation of journal items which a linked to different relations (read Partner-Id’s). This constrain should be removed. Often, reconciliation on items from different relations is highly desired.
> That feature is quite risky, it increases the risk of mistake. Moreover, it makes more complex the way the aged payable and receivable reports works. If you want to manage that, you should use a miscellaneous entries. 

Bank reconciliation receivables and payables:
Odoo does not allow the include both receivable journal items as well as payable journal items to reconcile a bank transaction. This constrain should be removed.
> I need to verify but I think it's not legal in all the countries. 

Sepa direct debit file:
Create Sepa-direct-debit file based on the selected customer's invoices. At the moment of file creation, the payment-status of the customer's invoices which are included in the direct-debit file should be changed to “included in direct debit”. The sepa-batch-reference (is payment reference together with invoice number) should be registered on the invoices included.
> Agree, I'll take your feedback into accounts for future improvements of the SEPA Direct Debit. 

Overdue report:
A downloadable Excel report presenting a list of all Relations (res.partners) with all its outstanding (overdue) amounts (=balance) of each invoice-numbers as being accumulated for each overdue period: for example periods of 0-30 days, 31-60 days, 61-90 days, etc. At this moment, Odoo already delivers a kind-off report, called aged report receivables. However, this report is a dynamic report, what implies that you need the open first all the relations before you have the whole list you want.
> This issue will be solved for the V11. You can already test it on the Master Version. There is now an option to unfold all lines. 

Follow-up process of overdue invoices:
In addition to the already available functionality of follow-up, we like to see improvement in the way follow-up messages are being selected to send out. We should be able to make a selection on the open-invoices list to include invoices in a follow-up batch-messages to send out together with the customer statements. The statement for a particular relation should include the selected invoice as well as the number of times invoices has been already being included (in the past) in the follow-up message. Therefore, it should we required to register the number of follow-ups being send-out on the invoice record.
> There are some ongoing improvements on the follow-ups. I think you should retest this feature once this will be done to check if this is still needed. 

Follow-up statement:
we need to have the list of overdue invoices together with the original invoice amount and balance amount (= invoice amount deducted with registered payments). At the moment, the invoice amount is not being presented on the statement.
> Why is it important for you to see the original invoice amount? In which use case do you plan to use that information while on the customer statement? (note also that you can switch on the invoice in one click). I'm not against the idea but as we are adding more and more info in that table, we need to carefully think about what should be there and what should not, to avoid having a crowded and unusable report. 

Send digital suppliers’ invoices (PDF) to Odoo.
It should be good when users are able te send supplier-invoices directly to Odoo (similar to the expense-functionality). So when a suppliers’ invoices is being received by Odoo, it creates an concept-invoice with the PDF attached. Now the concept can be amended with the correct data and confirmed.
> That's an interesting new feature. I'll add this in our backlog however I can't guarantee you that it will be done for sure. 

ICP rapport:
For the selected period, the report represents all the sums for all amounts of EU-sale transactions accumulated for each VAT-number. The amount of sale transactions are being split in sum of amounts representing products and sum of amounts representing services.
> Same answer than for Erwin, I think this should be included in the second iteration regarding the improvement of l10n_nl.

**Comment by Yenthe666** [comment]
For all you guys: There is a PR for this now. See https://github.com/odoo/odoo/pull/18636

**Comment by pedrobaeza** [comment]
> Reconciliation of journal items in regards to “relations”:
>In the functionality of reconciliation of journal items, Odoo does not allow reconciliation of journal items which a linked to different relations (read Partner-Id’s). This constrain should be removed. Often, reconciliation on items from different relations is highly desired.
>>    That feature is quite risky, it increases the risk of mistake. Moreover, it makes more complex the way the aged payable and receivable reports works. If you want to manage that, you should use a miscellaneous entries.

Indeed, that mustn't be allowed. It must be performed through an intermediate journal entry which transfers the balance from one partner to another. Maybe an action to perform this in standard would be good, or an specific option on reconciliation widget, but not directly. 

> Bank reconciliation receivables and payables:
> Odoo does not allow the include both receivable journal items as well as payable journal items to reconcile a bank transaction. This constrain should be removed.
>>    I need to verify but I think it's not legal in all the countries.

No, it's not legal in several countries (in Spain is restricted according several conditions), and again, it should be done with an intermediate journal entry. Check the module https://github.com/OCA/account-financial-tools/tree/8.0/account_netting that makes this. Include this feature in core would be great, but at least restricted through a security group.

@Reddconnect-Maarten IMHO, you are mixing specific localization requests with general accounting ones. This issue is for localization specifics. 

@lap-odoo maybe it's better to open a generic accounting improvements issue for the rest?

**Comment by lap-odoo** [comment]
@pedrobaeza @Reddconnect-Maarten : Indeed, it would be easier to only discuss country-specific issues on the l10n_xx issues. I've just created the following issue for the rest : https://github.com/odoo/odoo/issues/18637. 

Thanks

**Comment by smetl** [comment]
The first part of the Dutch Localization is finally ready and available on runbot:
http://252840-master-l10n-nl-reports-2-las-a4d770.runbot11.odoo.com/web/login
logs: admin/admin

It contains:

**New tags:** New account tags are available to have a more detailed P&L. The COA has been updated accordingly.
**P&L:** A new Profit & Loss report is available under the Netherland section is Accounting -> Reporting.
**Audit Report:** The audit report (.xaf file) can be generated from the General Ledger report by clicking on 'Export (XAF)'.
**Intrastat Report (ICP):** Available on Accounting -> Reporting.

Don't hesitate to inform me about anything that could be improved.

**Comment by ploegvde** [comment]
Wil je anders een interne taak maken voor jezelf. Dank je.

------------------------------

[image: Odoo Experts] <https://www.odooexperts.nl/>

Met vriendelijke groet,

Erwin van der Ploeg
06 - 14 60 12 42

Odoo Experts
088 - 377 40 00
Vredeweg 1s
1505 HH Zaandam
https://www.odooexperts.nl

[image: Twitter] <https://twitter.com/odooexperts> [image: Facebook]
<https://www.facebook.com/Odooexperts> [image: Google +]
<https://plus.google.com/+OdooexpertsNl> [image: LinkedIn]
<https://www.linkedin.com/company/odoo-experts> [image: Youtube]
<https://www.youtube.com/c/OdooexpertsNl>

Klik hier voor het laatste blog van onze Odoo Experts
<https://www.odooexperts.nl/blog/alle-blogs-3/?utm_source=email&utm_medium=handtekening&utm_campaign=email%20banner>


2017-08-09 14:57 GMT+02:00 Laurent Smet <notifications@github.com>:

**Comment by krijn1981** [comment]
The dutch P&L looks more into what the dutch customer expect. (many thanks for this).

What are we still missing?
**Anglo-Saxon Accounting G/L "goods to invoice account"**
In the past we saw the every quant within a movment is making a financial transaction.
This can lead to multiple transactions. 
Goods to invoice
a/ Stock
When the invoice is created the this will be counterentried as in.
COGS
a/ Goods to invoice

The G/L "Goods to invoice" should be more easiely being reconciled.

**Register payment**
The manual payment on a invoice, directly makes a entry receiveable/payable to bank.
This will give issues when a period will be closed and the bank will bring this payment (or receivement) in the next period. Untill the money is not paid or received it should "wait" on a temperory G/L.

**Multi currency - Reconcile**
Multi currency will suggest you can use more than 2 currency.
The following situation was not possible in previous versions but is very common.
I have a base currency in EUR (also reporting) currency called.
I have a invoice in GBP. I want to registrat this invoice also in the local currency.
But I will paid this invoice from my USD account.
Reconcile of this should be possible.

**Multi currency - reporting**
For local goverments is wish full to report in there local currency.
For example a company her mothercompany in The Netherlands, but the daughter company is placed in Japan. The company in Japan needs to follow the main currency of the Holland for reporting purposes.
But needs to local report in to the Japanees currency.
Wish full would be that you can set the reporting currency on a report.
every transaction needs to be recalculated based on currency and posting date of the journal entry.

** Contrains on sub administrations **
One check that a accountant always need to to if a G/L and a sub administration are the same.
Basic rule of administration. 
1. So on a receiveable and payable account there should be constrain to not allow any booking with a partner.
2. The same should be applied to stock accounts. Stock account versus a product.
3. On a bank and/or cash account you should not be allowed to book direct on the account only by setted journal. You als can say a cash/bank g/l account should always have a connected journal.

** Costing spreading (/Permanence) **
Costing spreading earlier maded by Onestein should also be in standard of Odoo. Due to Angelsaxon Accounting taught we would like to spread cost in there correct period for good match of costs and renvue.

** Report of goods movement versus financial impact **
Currently under certain situations goods movements are not delivering and financial facts.
To give better understanding to a financial department they should a overview/report where they easly can check the movements versus financial impact.

** Follow up header and footer **
In previous version we found out the header and footer of the follow up we quite hard programmed.
When you create a custom header and footer (by inherit the old ones) they will come to the follow ups.

** Purchase a service **
The expenses of a service are being financial booked (to COGS) when an invoice is validated.
Actually you should when the service should be done, based on that date the COGS and to receive invoices should be booked (based on PO value).

** Manufacoring **
The manufacuring module only makes a journal entry based on the goods movement. In this way we can not at value to a product that is not defined a stockable/cusomeable product.
The value of a labor of people is not added aswell the value partional value of the use a the machine.
See example "https://www.accountingcoach.com/blog/what-is-a-fixed-cost".

** Expenses summary **
A expense summary (2 or more) will lead to 2 or more journal entries.
The approval is for the whole summary so it should make 1 journal entry with 1 x payable and xxx time expenses.

** Payment order **
Processing a payment order will cost a finance people to much time.
I see it's already being changed in the runbot, but you still missing is a temporary overview to let a board member or the CEO to sign off the payments. 
Currently you can not easiely say this invoice i want to pay the order not.
The module https://www.odoo.com/apps/modules/10.0/account_payment_order/ 
will approach the reality of company beter.

** Cash flow prognose **
In V9/V10 we are missing a Cash flow calculation for the future.
Seeing what we have in the nearby furture in essential for a company. 

Regarding "2. Bank reconciliation receivables and payables:" of Maarten of Reddconnect.
The constrain should not be removed. It should split the journal entry. In Holland your are allowed to reduce the payable with receivable invoice. In other countries you need to ask permission to do that.

I hope by this we can contribute. If something unclear or there othter suggestions please contact me.

**Comment by krijn1981** [comment]
almost forgat

** Goods versus way of invoicing **
Normally a customer dictates how the invoice shall be done.
Based on delivery (after invoicing)
or based on ordering (invoicing before delivery).
If a customer is customer that has a bad reputation of payment, from finance perspective I want to receive money before the goods are sended. In the currenctly the product provides us the method of invoicing.
this means of one and the same product (in the stock) I need to create several products because not the situation but the product determines how to invoice. In V7/8 you also had a nice communcity feature called "Block delivery" in that way Finance had control over the out going goods without a payment on tricky customers. Its beter to receive the money than starting a legal procedure (€€€€€€€?).

**Comment by lap-odoo** [comment]
@krijn1981 : Hi Krijn, my answers are below

The dutch P&L looks more into what the dutch customer expect. (many thanks for this).
> Thanks ! Did you check it in details ? No error according to you ? 

What are we still missing?
Anglo-Saxon Accounting G/L "goods to invoice account"
In the past we saw the every quant within a movment is making a financial transaction.
This can lead to multiple transactions.
Goods to invoice
a/ Stock
When the invoice is created the this will be counterentried as in.
COGS
a/ Goods to invoice
The G/L "Goods to invoice" should be more easiely being reconciled.
> This is generic and not specifically linked to l10n_nl, can I ask you to report that on that generic issue please ? https://github.com/odoo/odoo/issues/18637. 

Register payment
The manual payment on a invoice, directly makes a entry receiveable/payable to bank.
This will give issues when a period will be closed and the bank will bring this payment (or receivement) in the next period. Untill the money is not paid or received it should "wait" on a temperory G/L.
> Same as above. Note that you are only suppose to register the payment on the invoice when you've really received the money. 

Multi currency - Reconcile
Multi currency will suggest you can use more than 2 currency.
The following situation was not possible in previous versions but is very common.
I have a base currency in EUR (also reporting) currency called.
I have a invoice in GBP. I want to registrat this invoice also in the local currency.
But I will paid this invoice from my USD account.
Reconcile of this should be possible.
> We've just finished a development that solves that issue. You can test it in Master if you want. So this is solved for V11. 

Multi currency - reporting
For local goverments is wish full to report in there local currency.
For example a company her mothercompany in The Netherlands, but the daughter company is placed in Japan. The company in Japan needs to follow the main currency of the Holland for reporting purposes.
But needs to local report in to the Japanees currency.
Wish full would be that you can set the reporting currency on a report.
every transaction needs to be recalculated based on currency and posting date of the journal entry.
> Can you report it on https://github.com/odoo/odoo/issues/18637 ? Thanks. It's generic and I want to make sure that partners not following l10n_nl can also follow that discussion. 

** Contrains on sub administrations **
One check that a accountant always need to to if a G/L and a sub administration are the same.
Basic rule of administration.

So on a receiveable and payable account there should be constrain to not allow any booking with a partner.
The same should be applied to stock accounts. Stock account versus a product.
On a bank and/or cash account you should not be allowed to book direct on the account only by setted journal. You als can say a cash/bank g/l account should always have a connected journal.
> same as above. I'm not sure to understand this. What concrete change do you want to see ?  

** Costing spreading (/Permanence) **
Costing spreading earlier maded by Onestein should also be in standard of Odoo. Due to Angelsaxon Accounting taught we would like to spread cost in there correct period for good match of costs and renvue.
> Same as above. You'll see we are discussing a lot anglosaxon accounting on the other task. I think we will probably create an issue to discuss only that subject. 

** Report of goods movement versus financial impact **
Currently under certain situations goods movements are not delivering and financial facts.
To give better understanding to a financial department they should a overview/report where they easly can check the movements versus financial impact.
> Same as above. That's a nice to have feature, not in our priorities right now. 

** Follow up header and footer **
In previous version we found out the header and footer of the follow up we quite hard programmed.
When you create a custom header and footer (by inherit the old ones) they will come to the follow ups.
> Same as above. There is an ongoing task to improve the followup feature. 

** Purchase a service **
The expenses of a service are being financial booked (to COGS) when an invoice is validated.
Actually you should when the service should be done, based on that date the COGS and to receive invoices should be booked (based on PO value).
> Same as above. 

** Manufacoring **
The manufacuring module only makes a journal entry based on the goods movement. In this way we can not at value to a product that is not defined a stockable/cusomeable product.
The value of a labor of people is not added aswell the value partional value of the use a the machine.
See example "https://www.accountingcoach.com/blog/what-is-a-fixed-cost".
> We indeed plan to improve that. 

** Expenses summary **
A expense summary (2 or more) will lead to 2 or more journal entries.
The approval is for the whole summary so it should make 1 journal entry with 1 x payable and xxx time expenses.
> That's not the case anymore, you can test it in Master. 

** Payment order **
Processing a payment order will cost a finance people to much time.
I see it's already being changed in the runbot, but you still missing is a temporary overview to let a board member or the CEO to sign off the payments.
Currently you can not easiely say this invoice i want to pay the order not.
The module https://www.odoo.com/apps/modules/10.0/account_payment_order/
will approach the reality of company beter.
> Same as above, that's generic. It makes sense to improve that. I think we need to review the SEPA and the checks also. This could be done as well. 

** Cash flow prognose **
In V9/V10 we are missing a Cash flow calculation for the future.
Seeing what we have in the nearby furture in essential for a company.
> Same as above. It's a nice to have feature. I'm sure it'll be done at some point but we first would like to improve the localizations (CoA, reports, taxes ...) 

Regarding "2. Bank reconciliation receivables and payables:" of Maarten of Reddconnect.
The constrain should not be removed. It should split the journal entry. In Holland your are allowed to reduce the payable with receivable invoice. In other countries you need to ask permission to do that.
> Yes, we agree on that. We don't want to remove this constraint because it's not legal in many countries. And where it's legal it can be managed using an intermediary account.  

** Goods versus way of invoicing **
Normally a customer dictates how the invoice shall be done.
Based on delivery (after invoicing)
or based on ordering (invoicing before delivery).
If a customer is customer that has a bad reputation of payment, from finance perspective I want to receive money before the goods are sended. In the currenctly the product provides us the method of invoicing.
this means of one and the same product (in the stock) I need to create several products because not the situation but the product determines how to invoice. In V7/8 you also had a nice communcity feature called "Block delivery" in that way Finance had control over the out going goods without a payment on tricky customers. Its beter to receive the money than starting a legal procedure (€€€€€€€?).
>I agree, we will discuss that internally.

**Comment by lap-odoo** [comment]
@ploegvde : Do you know if your colleague Julia will have the time to test the branch ? Thanks a lot

**Comment by ploegvde** [comment]
Yes we have planned to test today.

**Comment by ploegvde** [comment]
Runbot version is broken ;(

http://252840-master-l10n-nl-reports-2-las-a4d770.runbot11.odoo.com/web/login

**Comment by lap-odoo** [comment]
Should work with: http://253706-master-l10n-nl-reports-2-las-a4d770.runbot11.odoo.com/web/login 

Can you confirm ? Thanks

**Comment by ploegvde** [comment]
Yes is working. Tnx

**Comment by jvOrsouw** [comment]
The totals in the P&L report are not correct: EBITDA:  balance = NL_PCOST.balance - NL_PROFIT.balance
and NET result = NL_EBITDA.balance + NL_OTC.balance.
I've added 2 lines in the report to show the subtotal EBITDA and to make the correct Net result. I also deleted these terms from the 'cost' and 'other cost' lines, because a summary of the cost is not the same as the (sub)total of the result. 

I've made these changes now on runbot, but they need to be in the code too. 
Else I didn't find errors in the P&L report.

**Comment by lap-odoo** [comment]
@jvOrsouw : Thank you ! 

@smetl : Can you please update the code ? Thanks ;-)

**Comment by jvOrsouw** [comment]
Why is there still a check on country of a customer in the ICP report? Can we see the code for this report? 
The report seems to be doing what it is supposed to do, so no comments on that. 
But we would like all the numbers to be shown as rounded to whole euro's, because that is also the way the dutch tax authorities want the numbers. 

Does anyone know if the 'aangifte omzetbelasting' also has to be in whole euros? I think so, but I'm not sure.

**Comment by jvOrsouw** [comment]
1. On the XAF general ledger report: if the company info is not filled in completely, it generates an error a regular user cannot understand: 
<string>:16:0:ERROR:SCHEMASV:SCHEMAV_ELEMENT_CONTENT: Element '{http://www.auditfiles.nl/XAF/3.2}taxRegIdent': This element is not expected. Expected is ( {http://www.auditfiles.nl/XAF/3.2}taxRegistrationCountry ).
It would be very helpful if the error says that company information is missing, maybe even mention that 'country' has to be entered for this report to work.

2. Furthermore, the XAF-file is incomplete, there are no entries (transactions). 
3. And the periods all have the same number, that is also not correct. Suggested period number = last number of the year + 2 numbers for the month (701, 702, 703, etc.).

**Comment by yung-wang** [comment]
When you use an accounting date with for ex. vendor invoices, the Generic TAX eports shows the VAT in the wrong period /date.

You receive an invoice in the month aug, but you want the cost to be in the month jul. So the invoice date = 14-aug and the accounting date  = 14-jul.
When you look at the Tax report it shows the vat in jul.
According the dutch law, the VAT should always look at the invoice date if you use the Factuurstelsel:
https://www.belastingdienst.nl/wps/wcm/connect/bldcontentnl/belastingdienst/zakelijk/btw/btw_aangifte_doen_en_betalen/bereken_het_bedrag/hoe_berekent_u_het_btw_bedrag/factuurstelsel

Hope you can have a look at this.

By the way, i cannot login using admin/admin
http://253706-master-l10n-nl-reports-2-las-a4d770.runbot11.odoo.com/web/login

**Comment by krijn1981** [comment]
@ yung-wang,

I think i would a great benefit against other programs (they can you do this auto posting).
The fact there a 2 types of systems in the Netherlands, know as:

1. Kas stelsel
2. Factuurstelsel

Where you speaking about is Factuurstelsel. So we also should somewhere a boolean which system is used.
Note we also need to make a transit account to transfer the VAT to a the other period.

Normally we transfer this kind of things manually by a mem-journal.

**Comment by yung-wang** [comment]
@krijn1981 
Thanks for the feedback. I agree with you that to make this functionality complete we need to have kasstelsel too. But I think there is already an option in de accounting settings (Allow Tax Cash Basis).

Its a good workaround to book the VAT to an another period, but most of the accountants are not happy to make manual entries. So I think it is best that the report look at the right invoice date if it is "Factuurstelsel".

**Comment by jvOrsouw** [comment]
@lap-odoo : can you give us a new branch to test on? The last one is also broken.
Factuurstelsel is the most commonly used in the Netherlands, and if odoo wants to be taken seriously as having a decent accounting module, taxes, accounts payable and receivable should always be posted automatically in the invoicing date. Only the expenses or revenue is posted in the accounting period. This all in case of factuurstelsel. I haven't tested this, I'm relying on the test results of yung-wang who found that this is not the case at the moment.

**Comment by lap-odoo** [comment]
@jvOrsouw I've just launched the rebuild of the branch. It should be ready in +/- 30 minutes. 

@smetl will take care of correcting the issues you reported. 

Regarding the difference between factuurstelsel and Kas stelsel, I'll have a look at this but if we do changes, it'll be done in a second phase. We first would like to merge the changes that have already been implemented.

**Comment by yung-wang** [comment]
@lap-odoo Just like @jvOrsouw I agree we should make sure that Factuurstelsel works perfectly in the Dutch localization and if you want Odoo to be taken seriously in the Netherlands. It is really a showstopper, because it is common practise to have the expenses / reveneu in a different accounting period. I think this is really a let down to wait for a second phase. I hope you can still fix this. Thanks for the effort.

**Comment by lap-odoo** [comment]
@jvOrsouw : Regarding the ICP, it only takes intra-communautaire operations, that's why there is a check on the country. What is the issue ? 

Also, for the XAF export, we are well able to see the transactions that have been made. Maybe it was due to a wrongly defined filter on the general ledger ? Thanks

We are working on the other issues and will let you know when there are solved.

**Comment by lap-odoo** [comment]
Regarding Factuurstelsel and Kasstelsel, from what I've read (here: https://www.succesvolboekhouden.nl/kasstelsel-en-factuurstelsel) : 
- Kasstelsel : this corresponds to cash basis accounting, which is supported in Odoo 
- Factuurstelsel : we record the income and expenses based on the invoice date, not based on the date at which they are paid. This is well how Odoo works in standard. So I don't really understand. Regarding the BTW, they say "De btw op ontvangen en verstuurde facturen geef je aan in de periode van de factuurdatum". Which is the standard behaviour if you don't change manually the accounting date.

**Comment by astirpe** [comment]
@lap-odoo Factuurstelsel: the issue occurs when you create an invoice with an accounting date of a period different than the one of the invoice date. In comment https://github.com/odoo/odoo/issues/18572#issuecomment-322170029 there is an example of use case.

**Comment by yung-wang** [comment]
@lap-odoo Thanks for researching the problem. I'm glad the Cash Basis is fulfilling the kasstelsel.

For the Netherlands it is very common to register the revenue and cost in another period. The solution that Odoo provides is using the 'accounting date' field. This is what Odoo advised me to use. But using this is exactly creating the problem.

Let's say you get an invoice for the rent of **July**. But the invoicedate = **17-Aug**. According the accounting rules of the Netherlands, is it best to register the cost in July.

So when you create a vendor invoice, you use invoicedate according the print = **17-Aug**. For the cost registering in period July you use accountingdate = for ex. **1-Jul**

If you run the VAT report for Aug, the VAT is empty. But if you do this for Jul, there you find your VAT. That is not correct according: 

> Regarding the BTW, they say "De btw op ontvangen en verstuurde facturen geef je aan in de periode van de factuurdatum".

So the VAT report is not useful in the Netherlands.

**Comment by lap-odoo** [comment]
@yung-wang Thank you for the explanation.

As explained by @krijn1981, a transfer account should then be used and two entries should be generated at the validation of the invoice (if Factuurstelsel is active and if the invoice date is different than the accounting date), right ? 

Let's take your example : 
Invoice Date = 17 August 
Accounting Date = 1 July 

Journal entry 1 July
D                                           C
Expense Account                  Expense Transfer Account

Journal entry 17 August
D                                           C 
Tax                                         Account Payable
Expense Transfer Account

Is that correct? If yes, what should be the account used as "expense transfer account"?

**Comment by yung-wang** [comment]
@lap-odoo Hi Laura, the solution by @krijn1981 is a workaround to use a transfer account. It's not how Dutch accountant usually book invoices.

In V10 the accounting module is changed to remove bookingperiods and it is now date based. Previously you could create an invoice with invoice date 17-aug and book it in period 2017/07. When I asked some consultants at Odoo, they propose to use accounting date instead bookingperiods. But that is a different discussion.

So in mine opinion, we shouldn't use transfer accounts. It's better to improve the VAT report. So when it is Factuurstelsel (an option in the Settings), the VAT report should use the invoice date for calculating the tax. (just like the solution in the OCA/l10n-netherlands)
If this is not possible and you want to use the transfer account solution, than in the Netherlands you can use something like 200000 Tussenrekening. But than again, you need a settings that you can configure which account is used. The accounting system is much more flexible than the one you use in Belgium. The customer should be able to select his own transfer account.

**Comment by fpodoo** [comment]
@yung-wang It's NOT POSSIBLE to report the VAT at the invoice date. Example: "Let's say you get an invoice with date July 31, but you received and record it in Aug 22. In this case, the invoice date must be July 31 (the one on the document) but you need to report in the tax of August as you probably already filed the July tax statement --> you can not modify a statement you already filed".

So, by definition, Odoo works that way:
- The invoice date: set by the vendor, the one on the invoice
- The accounting date: the period in which you want to report your costs AND taxes
- The Due Date: based on invoice date + payment term

We have tested Exact Online (version NL) and Exact Online do EXACTLY like in Odoo: two dates: invoice date & accounting date (cost+tax report). I think Exact is largely used in Netherlands, which makes us feel like the current Odoo behavior is the right one. Do you have an example of other Software that report the tax in a different period than the cost?

**Comment by jvOrsouw** [comment]
Afas profit reports tax based on the invoice date, as dutch law requires, even when the accounting period, thus the period of the expense/revenue, is not the same period.
If taxes have already been reported and an invoice is received for that period, the tax is reported the next time, or by a separate 'suppletie'-report, depending on the total amount of tax that was not reported in time. If this (reporting tax based on invoice date) is not possible in Odoo, we'll have to find a workaround. As Odoo is trying to comply with dutch law, this would be a great addition. I wouldn't call it a showstopper as many dutch companies are already using Odoo as it is, and apparently, there is more software available and widely used, that also doesn't comply with these laws.

**Comment by fpodoo** [comment]
@jvOrsouw Odoo and Exact actually comply with the law by requesting users to set the accounting date to the one you'd like to report your tax to.

**Comment by jvOrsouw** [comment]
That is a way of using the accounting date. But then you loose the possibility of reporting the expenses or revenue in the period they mention. Most dutch accountants are only looking for ways to have costs (and revenue) in the period the invoice mentions, and tax in the invoice period. And if the cost period is different from the invoice period, they'll use the accounting date to get the cost in the desired period. But we can also use your suggestion, combined with the use of deferred revenue and assets to get costs and revenue in the correct period. I think we have established that Odoo works differently than we had anticipated, but there's definitely room for everyone's method of choice.

**Comment by yung-wang** [comment]
Hi @fpodoo, thank you for your comment. I'm glad you mention Exact Online, since I used to work at Exact Software in a team responsible for introducing Exact Online to the Accountancy branche in the Netherlands. I will check this on Exact Online using a trial version. I also know that SAP R3 and AXI works exactly the way I described (previous work experience as an ERP consultant too)

Sorry to say this, but according the dutch tax authority, the invoice date is leading for the taxes according Factuurstelsel (see the links in earlier comments). So how Odoo works with accounting date is not correct. And if you receive the invoice too late, like in your example, you can register the tax in the next month manually or intelligently if your software provides it.

So to come back at Exact Online. I created a test case to try it your self. This is how most of the Dutch accountants/bookkeepers book there vendor invoices.
screenshot 1. Create invoice date 18-aug for 121,- inc TAX, but assign to period 2017/07 (JUL) just like we use Odoo accouting date. Process the invoice
screenshot 2. Check the P&L for periode 07 and you see the cost 100,-. When you check periode 08, it is empty
screenshot 3. Create a TAX report for Aug and you will see the correct amounts. When you check Jul it is empty. This is how Factuurstelsel should work.
screenshot 4. Just to show you the table of period.
I hope I showed you how to do Dutch Accounting. As you can see, Exact Online is doing this correctly. Odoo with Accounting date is wrong.

@jvOrsouw, I only agree that this is not a showstopper if we implement Odoo by a customer since we have workaround and other modules to fix this. But here we are talking about creating the Dutch Localization. Since that is the case, this is a showstopper. Odoo should at least support the Factuurstelsel correctly, not using workarounds. Like Odoo I want to keep this K.I.S.S. and believing in Odoo motto "Fast and Simple". That is why I want to help as much as possible.

I would love to see that Odoo can comply more to the Dutch localization and with opening this issue I'm happy you give that opportunity. I'm very happy how good Odoo works and the software is very promising, but the localization is still lacking for The Netherlands. For now we are using mostly custom modules or workaround to fill this Gap. 
So I hope you can fix this. But the accounting date should be avoided for the standard TAX report in the Netherlands.

Can we also continue on the other open items? Do we still need to check:
New Tags
P&L
Audit Report
ICP

![screen_04](https://user-images.githubusercontent.com/30628967/29506991-7c90f0d0-864e-11e7-9a8d-ac7f831fe2ad.png)
![screen_03](https://user-images.githubusercontent.com/30628967/29506992-7c927586-864e-11e7-830d-b27fc2ac6197.png)
![screen_02](https://user-images.githubusercontent.com/30628967/29506994-7c980dde-864e-11e7-9d52-f75631067786.png)
![screen_01](https://user-images.githubusercontent.com/30628967/29506993-7c940964-864e-11e7-9459-5720ea2a051e.png)

**Comment by smetl** [comment]
New database available:
http://256282-master-l10n-nl-reports-2-las-70f534.runbot11.odoo.com/?db=256282-master-l10n-nl-reports-2-las-70f534-all
logs: admin/admin

Regarding your comments, I fixed the code:

- P&L: 
* Totals have been fixed by adding a new section for the EBITDA.
- XAF: 
* Error regarding the company data not filled fixed.
* Periods number are now unique.

**Comment by smetl** [comment]
**/!\ The branch will be merged at the end of the next week.**

Thank you very much to test it before this date, otherwise I will not have time to fix the issues regarding your comments.

**Comment by jvOrsouw** [comment]
@smetl : last database is broken, could you rebuild? Then I can give it a quick last test.

**Comment by smetl** [comment]
@jvOrsouw I do this asap but runbot is a bit capricious. Thank you for your patience.

**Comment by smetl** [comment]
@jvOrsouw You can test on the last runbot build even if it's red because it's not related to a bug with the Netherlands localization:
http://260006-master-l10n-nl-reports-2-las-5724cb.runbot9.odoo.com/?db=260006-master-l10n-nl-reports-2-las-5724cb-all

**Comment by jvOrsouw** [comment]
I get an error when trying to install dutch COA: 
- created a new company
- added access rights for new company to admin, and switched to new company
- tried to install new COA through accounting - settings - choosing 'NL grootboekschema' and get below error.

Fout:
Odoo Server Error

Traceback (most recent call last):
  File "/home/odoo/runbot/extra/runbot/static/build/260006-master-l10n-nl-reports-2-las-5724cb/odoo/http.py", line 648, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/runbot/extra/runbot/static/build/260006-master-l10n-nl-reports-2-las-5724cb/odoo/http.py", line 309, in _handle_exception
    raise exception
KeyError: 2

**Comment by jvOrsouw** [comment]
I get an error trying to run the ICP report: 

Fout:
Odoo Server Error

Traceback (most recent call last):
  File "/home/odoo/runbot/extra/runbot/static/build/260006-master-l10n-nl-reports-2-las-5724cb/odoo/http.py", line 648, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/runbot/extra/runbot/static/build/260006-master-l10n-nl-reports-2-las-5724cb/odoo/http.py", line 309, in _handle_exception
    raise exception
AttributeError: 'module' object has no attribute 'items'

**Comment by jvOrsouw** [comment]
[ToelichtingAuditfileFinancieel3.2.pdf](https://github.com/odoo/odoo/files/1270256/ToelichtingAuditfileFinancieel3.2.pdf)
[XmlAuditfileFinancieel3.2_FunHie.pdf](https://github.com/odoo/odoo/files/1270257/XmlAuditfileFinancieel3.2_FunHie.pdf)

I'm still missing the transactions lines in the XAF auditfile. It only lists the number of transaction lines, the total debit and credit, and then lists the journals, but there is no detailed information of the transactions (transaction date, transaction number, general ledger & amount debit or credit). 
Opening Balance			0..1, F
		Opening Balance Line	0..*, F
		Subledgers				0..1, F
			Subledger			0..*, F
				Subledger Line	0..*, F
	Transactions				0..1, F
		Journal					0..*, F
		Transaction				0..*, F
			Transaction Line	0..*, F
				VAT				0..99, F
				Currency			0..1, F
	Subledgers					0..1, F
		Subledger				0..*, F
			Subledger Line		0..*, F
				VAT				0..99, F
				Currency			0..1, F

example:
-<transactions>

<linesCount>1</linesCount>

<totalDebit>3.14</totalDebit>

<totalCredit>3.14</totalCredit>


-<journal>

<jrnID>String</jrnID>

<desc>String</desc>

<jrnTp>B</jrnTp>

<offsetAccID>String</offsetAccID>

<bankAccNr>String</bankAccNr>


-<transaction>

<nr>String</nr>

<desc>String</desc>

<periodNumber>1</periodNumber>

<trDt>1967-08-13</trDt>

<amnt>3.14</amnt>

<amntTp>C</amntTp>

<sourceID>String</sourceID>

<userID>String</userID>


-<trLine>

<nr>String</nr>

<accID>String</accID>

<docRef>String</docRef>

<effDate>1967-08-13</effDate>

<desc>String</desc>

<amnt>3.14</amnt>

<amntTp>C</amntTp>

<recRef>String</recRef>

<matchKeyID>String</matchKeyID>

<custSupID>String</custSupID>

<invRef>String</invRef>

<orderRef>String</orderRef>

<receivingDocRef>String</receivingDocRef>

<shipDocRef>String</shipDocRef>

<costID>String</costID>

<prodID>String</prodID>

<projID>String</projID>

<artGrpID>String</artGrpID>

<workCostArrRef>String</workCostArrRef>

<qntityID>String</qntityID>

<qntity>1</qntity>

<bankAccNr>String</bankAccNr>

<bankIdCd>String</bankIdCd>


-<vat>

<vatID>String</vatID>

<vatPerc>0</vatPerc>

<vatAmnt>3.1</vatAmnt>

<vatAmntTp>C</vatAmntTp>

</vat>


-<currency>

<curCode>AED</curCode>

<curAmnt>3.14</curAmnt>

</currency>

</trLine>

</transaction>

</journal>


-<subledgers>


-<subledger>

<sbType>CS</sbType>

<sbDesc>String</sbDesc>

<linesCount>1</linesCount>

<totalDebit>3.14</totalDebit>

<totalCredit>3.14</totalCredit>


-<sbLine>

<nr>String</nr>

<jrnID>String</jrnID>

<trNr>String</trNr>

<trLineNr>String</trLineNr>

<desc>String</desc>

<amnt>3.14</amnt>

<amntTp>C</amntTp>

<docRef>String</docRef>

<recRef>String</recRef>

<matchKeyID>String</matchKeyID>

<custSupID>String</custSupID>

<invRef>String</invRef>

<invPurSalTp>P</invPurSalTp>

<invTp>C</invTp>

<invDt>1967-08-13</invDt>

<invDueDt>1967-08-13</invDueDt>

<mutTp>I</mutTp>

<costID>String</costID>

<prodID>String</prodID>

<projID>String</projID>

<artGrpID>String</artGrpID>

<qntityID>String</qntityID>

<qntity>1</qntity>


-<vat>

<vatID>String</vatID>

<vatPerc>0</vatPerc>

<vatAmnt>3.14</vatAmnt>

<vatAmntTp>C</vatAmntTp>

</vat>


-<currency>

<curCode>AED</curCode>

<curAmnt>3.14</curAmnt>

</currency>

</sbLine>

</subledger>

</subledgers>

</transactions>

**Comment by jvOrsouw** [comment]
P&L report looks very nice! 👍

**Comment by smetl** [comment]
@jvOrsouw Thank you for your review. I fixed the troubles on the ICP report.

**Comment by smetl** [comment]
@jvOrsouw I have a new build ready for testing:
http://261077-master-l10n-nl-reports-2-las-df2882.runbot8.odoo.com/web/login

Thank you very much!

**Comment by lap-odoo** [comment]
The improvements on l10n_nl have been merged in branch SaaS 18. 
So, do not hesitate to test and to report any issue. 

Thanks for your help

**Comment by ploegvde** [comment]
I see that when installing the NL app we also get US GAAP reports. That seems odd? In the section 'Netherlands' is no Balance report anymore. And that we need. @lap-odoo Can you check.

**Comment by ploegvde** [comment]
Ok, but isn't in confusing for the enduser that there are now 2x P&L. 1 for Netherlands and 1 generic. If NL localisation is installed i would rader see the generic hidden. It is confusing.

For the balance sheet, we will collect the information for you.

**Comment by jvOrsouw** [comment]
I get an error when trying to run the XAF auditfile:
![image](https://user-images.githubusercontent.com/19901265/30955454-fccef30c-a433-11e7-8f23-beb274761fcd.png)
apparently, something is not quite right with rounding the numbers.

**Comment by lap-odoo** [comment]
@jvOrsouw Thanks Julia, @smetl will take care of fixing this

@ploegvde : Well, as currently we don't have a specific BS and P&L for each accounting localization, I think it makes sense to always show the generic reports too.  Moreover, we have clear submenus (the first one with the localization specific reports and then the generic ones). We will see what's the feedback from the users and change that if needed.

**Comment by jvOrsouw** [comment]
Hi, 
Again, I have an issue with the XAF-report. Tested it today on saas-18. The suppliers and customers are not mentioned: 
 <country>NL</country>
                </streetAddress>
                <customersSuppliers>
                    
                </customersSuppliers>
                <generalLedger>

Another small issue is the company-street. With the new split of the street into streetname, house-name and door-name, the fields for 'street' and 'number' are not filled correctly. 
Can it be changed that only 'street' is shown, as a combination of the new fields, and in the correct format for the Netherlands?: 

Street Format | %(street_name)s %(street_number)s%(street_number2)s

**Comment by smetl** [comment]
Hi,

@jvOrsouw I'm doing a branch with the right street formatting. About the `<customersSuppliers/>`, I don't get the problem. I tested some XAF exports, every partner is present with the tag `<customerSupplier/>`, Can you give me the url of the runbot having this issue plz? Thanks!

**Comment by jvOrsouw** [comment]
@smetl I've been testing on saas-18:
http://268826-saas-18-8d4b9b.runbot11.odoo.com/web#action=661&menu_id=228

**Comment by smetl** [comment]
@jvOrsouw They are not present in the XAF report because they are declared with another company "YourCompany". Is it wrong?

**Comment by jvOrsouw** [comment]
invoices were created in company 'nederland'


[image: Odoo Experts] <https://www.odooexperts.nl/>


Met vriendelijke groet,

Julia van Orsouw
06 - 28 43 20 82

Odoo Experts
088 - 377 40 00
Vredeweg 1s1505 HH Zaandam
https://www.odooexperts.nl

[image: Twitter] <https://twitter.com/odooexperts> [image: Facebook]
<https://www.facebook.com/Odooexperts> [image: Google +]
<https://plus.google.com/+OdooexpertsNl> [image: LinkedIn]
<https://www.linkedin.com/company/odoo-experts> [image: Youtube]
<https://www.youtube.com/c/OdooexpertsNl>

Klik hier voor het laatste blog van onze Odoo Experts
<https://www.odooexperts.nl/blog/alle-blogs-3/?utm_source=email&utm_medium=handtekening&utm_campaign=email%20banner>



2017-10-02 14:43 GMT+02:00 Laurent Smet <notifications@github.com>:

**Comment by jvOrsouw** [comment]
there is a sort of company-lock on the partners, on the sales&purchase tab. If I remove it, or set it to company 'nederland', the partners appears on the XAF-report. I see the company is automatically assigned when a partner is created. I understand the multi-company use. So for regular users, they only see the partners of the current company. So it only could be a problem when a user uses the admin-account in a multi-company setting. I guess then no change is needed on that part.

**Comment by ploegvde** [comment]
There is a problem in the intrastat ICP. Transport costs are added a product lines in sale orders. These products are set as type 'service' so that the transport is not created i as delivery order line. It is a service, but for ICP it is part of the product price. Or you include the transport in the sale price or you leave it out and invoice separate, but product + transport is the total exported amount.

On the ICP report now the transports service is set in the 'service' column of the report. That is not ok and it needs to be in the 'delivery' column.

So a transport 'service' product is not a service but a delivery. So this is a problem.

Maybe it is better to split 'delivery' and 'service' not based on product type but on taxcode? There are in the Dutch accounting already separate taxes for delivery and services.

**Comment by lap-odoo** [comment]
@ploegvde Hi Erwin, I agree this could be improved but a quick fix is simply to change the delivery product into a consumable (instead of service).

**Comment by ploegvde** [comment]
@lap-odoo The problem with changing to consumable, but then the warehouse people get all the transport cost lines in there picking and that is unwanted and confusing for these employees. Als a negative stock on transport cost looks strange.

**Comment by lap-odoo** [comment]
@ploegvde I understand it's not the ideal, but it's good workaround for now. 

Would you be able to provide me the list of taxes and, tax per tax, define if it should be in the product or service column ? We will update the code accordingly.

**Comment by lap-odoo** [comment]
We are now ready to start the second phase of development for the Dutch Localization. 

What do you suggest ? 
- Specific Balance Sheet Report ? 
- SBR Reports ? 

I'm waiting for your suggestions and detailed information. 

Thank you

**Comment by ploegvde** [comment]
What i think is most important is how can we help our V10 customers with the legally required ICP report?

**Comment by yung-wang** [comment]
Hi Laura, do you know if any of these localization will be available for the community version? Because since v11 the bookkeeping functionality is hidden in the community. So we are working here for the localization (to improve the requirement by law) I hope this will be also available in the community version.

**Comment by jvOrsouw** [comment]
I would like to see that SBR-reporting is implemented. It will become mandatory to report according to SBR-standards, and following is a direct link to dutch autorities to 'auto-upload' f.e. dutch tax-report.

**Comment by lap-odoo** [comment]
@yung-wang : Hi, 

Starting from Odoo Version 11, we decided to make a clearer distinction between the Invoicing and the Accounting applications. 

The Invoicing application is part of the Community version while the Accounting application is only supported officially (= by Odoo) in the Enterprise version. 

This doesn't mean the Accounting cannot be managed on the community version, but this means that it is not included in the standard community version (with only the official Odoo modules). Indeed, we've kept the entire accounting engine and most meta-data in the Odoo Community code so that it's still possible for third-party community apps to implement accounting features based on the official Odoo code (by unhiding accounting features for example).

In short, Odoo SA does not support the accounting features of Odoo Community, but it's ok if third party apps do it (and we don't make accounting developer's life complex).

In short, all the improvements that are done on the invoicing part OR on the CoA, taxes, fiscal positions, ... will impact the community version. However, the reports are only for the enterprise version. 

I hope this clarifies the situation

**Comment by yung-wang** [comment]
@lap-odoo Hi Laura, thank you for your explanation. We will continue to work to improve the localization in the Community version. For us it will be difficult to maintain both version of the localization and it will be a duplication of our effort. But if you want some suggestion I would say:

- VAT could be improved (see this    https://www.odoo.com/apps/modules/10.0/l10n_nl_tax_statement/) The Omzet and BTW should be printed next to each other.
- Factuurstelsel should be reflected in the VAT report (see this   https://github.com/OCA/l10n-netherlands/pull/108)
- I agree with the above comments about ICP report and SBR / XBLR

**Comment by ploegvde** [comment]
Hello, about the splitting of services and products. I'm curious what others think would be best solution. My proposal to use "service tax codes" can work, but then you vane to set the product type and tax correct on the product and that can be forgotten. @yung-wang What would you suggest?

**Comment by yung-wang** [comment]
Hello Erwin @ploegvde , we still need to split the services and products because the Dutch Authority makes it very difficult but also flexible. A customer can do a seperate ICP for products but also for services and they can do the services for 1 month and the products for 1 quarter. The customer is also allowed to combine them both into 1 ICP. So because of this we need to keep them split. But I'm not sure how you would configure the "service tax codes", it might work.

If you have a customer that only does services, than I would just create only tax codes for services and archive the products ones.

> https://www.belastingdienst.nl/wps/wcm/connect/bldcontentnl/belastingdienst/zakelijk/btw/zakendoen_met_het_buitenland/goederen_en_diensten_naar_andere_eu_landen/opgaaf_icp/tijdvak_opgaaf_icp/
U levert goederen én diensten
Als u zowel goederen als diensten levert, kunt u apart opgaaf ICP doen over de goederen en de diensten. Dat hoeft niet in 1 opgaaf ICP. Zo kunt u bijvoorbeeld voor uw goederen per maand opgaaf ICP doen. En voor uw diensten per kwartaal. Maar u kunt uw goederen en diensten ook in 1 opgaaf ICP doen. U volgt dan de regels die gelden voor de opgaaf ICP over goederen.

**Comment by ploegvde** [comment]
@yung-wang That is not what i mean. The problem is that the ICP report splits the products and services based on product type. But eg 'transport cost' is a service that you can/have to report as product on the ICP report. Because 'transport cost' is a part of the cost of a product.

But now the transport cost appears as service on the ICP report.

So my suggestion is to not split the services and products based on product type, but based on TAX code. We have separate tax codes for service products @lap-odoo That are the tax codes with the word "Diensten"

This is the best solution i can come up with.

**Comment by yung-wang** [comment]
@ploegvde Ah ok, sorry I misunderstood.
In Exact Online they have an extra option in the VAT code, similar to what you suggest to let the ICP report look at the VAT code.

> https://help.exactonline.com/leg-nln/NL/All/Content/Financial/Taxes/ECSalesList/FinTax_CrtECT.htm

**Comment by ploegvde** [comment]
The problem is that best solution would be to create a setting on the tax code if it is for products or services. But on stable Odoo can not do that. So we can only do it on the available tax codes from the Dutch module. That means user can not change it or create there own tax.

So for now it is ok, but for next version maybe create a setting. "ICP Products", "ICP Services" or "Not on ICP"

Agree that we can ask Odoo do continue this way?

**Comment by yung-wang** [comment]
@ploegvde Im not sure what you're suggesting. Do I understand it correctly to keep things like it is for now? And for V12 we ask Odoo to add ICP settings in de TAX codes?

**Comment by ploegvde** [comment]
No, for V11 change the report based on the now existing taxes. hardcoded use the "dienste" tax codes for the services and the other tax codes for products. In V12 make a setting on the tax.

**Comment by yung-wang** [comment]
In the OCA localization we use only 1 tax code "Verkopen export binnen EU" that is linked to label "3b. Leveringen naar/diensten in landen binnen de EU (omzet)"
So im not sure how to do a hardcoded version with the existing codes if we only use 1 tax code. Do you have a different tax configuration? And I'm not really a big fan of hardcoded changes, but if that is the easy way, than that is OK. @lap-odoo what do you suggest?

For V12 that would be great to have extra settings on the tax codes.

**Comment by ploegvde** [comment]
You are right, i'm totally mistaken. For export we do not have separate tax code for services. So forget my solution.

**Comment by jvOrsouw** [comment]
So to sum up on improvement on taxes:
- 'aangifte omzetbelasting' to be based on invoice date, not accounting date
- separate taxes for export: export products, export services
- ICP report based on the new taxes, so the distinction between products and services is no longer based on the product type, but on the tax used in the transaction
- more flexible ICP report, possibility to only report products, or only report services
- possibility to make (tax) reports (ICP and 'aangifte omzetbelasting') static, so the user can save the report in Odoo and see what numbers were filed to the dutch tax authorities
- SBR reporting: file the tax-report digitally from Odoo to the dutch tax authorities.

**Comment by lap-odoo** [comment]
Ok, let's start with the ICP report first. 

We can use the tags on the taxes to determine if a line should be considered as product or as service (in the ICP report).

@ploegvde : You told me it's already possible to distinguish them based on the term "Diensten". Is that well correct ? So basically, I took the file in attachment, and if the tag contains "diensten", this should be considered as a service and not a product ? 

Thanks

**Comment by ploegvde** [comment]
Hi, i was mistaken and it can not be done based on tax code. I propose the introduction of a extra field where we can set the ICP product type. Default for consumable and stockable it is 'products' and for service it is service 'service'. But when needed you can set a consumable or stockable on 'service' for ICP. You can hide the setting in developer mode.

**Comment by yung-wang** [comment]
Back from Holiday.
@ploegvde Ok, no problem. Lets see if we can have an extra field to set the ICP product (tax) type. For me this sounds as the best solution.
@jvOrsouw That is a nice list for improving Odoo for the localization for The Netherlands.

**Comment by ploegvde** [comment]
I want to propose a new features to the tax reporting:

After the tax report is made the taxes needs to be posted to the tax authorities and we need a process of validating, posting and paying the tax.
For that the following things need to be done:
1.	The tax values needs to pushed (SBR/XBLR) to the Dutch Tax website/portal. 
2.	The journal items that were used to create this tax report needs to be set as done. So that we can distinguish lines that are already in report and which lines are new. So we can report in eg 2nd month extra lines from first month that where added afterwards (maybe extra option "also check previous periods of this fiscal year"
3.	Create a payment for the tax amount. This will clear all tax amounts and move the tax amounts to the account payable. That can be reconciled on bank statement.

**Comment by yung-wang** [comment]
@ploegvde Great proposal.

In the OCA module we already solved suggestion 2.
https://github.com/OCA/l10n-netherlands/pull/108

**Comment by ploegvde** [comment]
@lap-odoo I found a bug in the ICP report. When no product is set on the invoice line, the module can not determine if it is 'product' or 'service'. In that case there is a difference between the 4b tax group on the tax report (aangifte omzetbelasting) and the ICP report, which has the details of tax group 4b.
What we can do it add a column "unknown". Then the user can make up his mind if it is product or service when he makes the ICP reporting on tax authorities.

Can you propose a fix soon. It s really hard to find the differences in tax report and icp report. Thanks.

**Comment by lap-odoo** [comment]
@ploegvde Thanks for reporting this. 

@smetl will take care of fixing that. 

For the ICP Report, what we will do is to add two tags (on taxes) : "Product" and "Dienst". Based on the tag defined on the tax, it will go on the column product or the column diensten.

**Comment by ploegvde** [comment]
That won't be possible. There is only 1 tax group 4b and that is used for both products and services and also in most cases set by the fiscal position of the partner.  So you can not set tags on that. Or do i misunderstand.

**Comment by lap-odoo** [comment]
In that case you will have to duplicate the taxes and one is used for the services and the other for the products. That's what we have in the Belgian Localization. We also need to distinguish the product type when selling or buying outside belgium, so we use different taxes.

**Comment by ploegvde** [comment]
Ok, but then we also have to change the tax chart and tax report. We will have a tax group 4b1 for products and a 4b2 for services. On tax report the total of 4b is displayed. On ICP report we can split is.

That also solves the problem for services that needs to be on ICP report as product.

What does the rest think?

**Comment by lap-odoo** [comment]
@ploegvde Indeed, we will have to change the tax chart and the ICP report. I don't think the tax report will be impacted. 

That's already how we manage it for Belgium so at least we stay in the standard of Odoo.

**Comment by ploegvde** [comment]
Ok, how do we move on with this?

**Comment by lap-odoo** [comment]
@ploegvde 

First, we need to fix the V11. @smetl is taking care of this, he will inform you when it's done. Concretely, we will change the way the ICP report is generated, it will not be based on the product type anymore but based on tags. We will add two tags on taxes : "Product" and "Dienst". You will have to deduplicate the taxes to have one for producten and one for diensten. 

Then, for the next version (I hope for the next saas release), we will deduplicate the taxes (as it is done for l10n_be). So for each tax, one will apply on services and the other on taxes. For that part, I have some questions : do we need to deduplicate ALL the taxes ? Some of them contain "diensten" in their name, what does it mean ? Thanks to that, the ICP report will be correct and no additional configuration will be required for the end users.

**Comment by ploegvde** [comment]
we need to deduplicate ALL the taxes >> No for the ICP report we have only one tax and that is "Verkopen export binnen EU". That is for all deliveries/services to countries within the EU. So this one needs to be splitted into "Verkopen export binnen EU Producten" and "Verkopen export binnen EU Diensten" (where diensten means services).

But if we do that we also need to change the Dutch Fiscal position.

**Comment by lap-odoo** [comment]
@ploegvde But if you only duplicate the tax "Verkopen export binnen EU", indeed it will be problematic for the fiscal positions. So all the taxes that could potentially be replaced by "Verkopen export binnen EU Producten/Diensten" should be duplicated too

**Comment by ploegvde** [comment]
We will provide you input.

**Comment by influenist** [comment]
Guy (@ploegvde, @yung-wang @jvOrsouw), I want to raise your attention regarding the effort for input. Although as a Dutch potential Enterprise user of ODOO I love to see this is getting somewhere to comply with the local Dutch regulations, @lap-odoo released a statement regarding the removal of the accounting addon in the Spanish localization issue. Please see https://github.com/odoo/odoo/issues/18565#issuecomment-324546449

**Comment by lap-odoo** [comment]
@influenist I think I already answered that question. I suggest you to have a look at my comment from 26 days ago. If it's not clear or if you still have questions, do not hesitate to ask me. But I think most of the questions have already been asked and received a reply on the Github issue related to l10n_es that you reference.

**Comment by yung-wang** [comment]
@influenist If I understand it correctly, accounting will be still there. Since you might be using Enterprise, there is for sure accounting. So no need to worry. Odoo is only 'hiding' this in the community version, but it is still there.

**Comment by lap-odoo** [comment]
@ploegvde @jvOrsouw @yung-wang Could you please provide input regarding the taxes ? 

Which taxes do we need to duplicated (to distinguish the ones applied on services and the ones applied on products)? We also need to adapt the fiscal positions accordingly. 

Thanks for your input

**Comment by jvOrsouw** [comment]
Hi Laura,
In order to make this work, here is my proposal:
- duplicate tax 'verkopen export binnen EU' and name it 'verkopen export binnen EU (diensten)'. 
- New tax has same tax rules as the current Tax (same tag '3b', no need for a new tax account)
- change tax mapping of fiscal position 'EU landen': taxes starting with 'verkopen' and ending with 'diensten' get the new TAX to apply 'verkopen export binnen EU (diensten)'.  This is 4 lines in total.
- create a new fiscal position 'installatie en afstandsverkopen'. This will be a fiscal position for customers within the EU who don't have to report taxes. This new fiscal position has no tax mapping, because tax mapping is only necessary when a revenue-limit is exceeded. 
- ICP report looks at the Tax of the move.line to determine if the tax is 'diensten' (tax 'verkopen export binnen EU (diensten) is used), or products (tax 'verkopen export binnen EU' is used). So it takes all the move.lines with tag '3b leveingen naar/diensten in landen ....',and gathers the rest of the needed information from that startingpoint. 
- The new tax is linked to the same tag, no adjustments for the tax-report is needed. HOWEVER, if both '3b' taxes get a different tag (see also [this ](https://github.com/odoo/odoo/commit/676ae771a614475fe5fa6049a9008df784d3e0e3) issue) the Tax-report also has to be adjusted so both tags are accounted for. In the tax-report it should show as only 1 line, the total of the two tags.
- new report: total amount revenu for fiscal position 'installatie en afstandsverkopen' per country: By default it only looks at the current fiscal year. Should have the same options (default no analytic accounts) as a P&L / Balance sheet.

**Comment by lap-odoo** [comment]
@jvOrsouw Thanks a lot ! 

You will be informed when this is done so that you can test it. 

Also, the CBS report should soon be ready for testing, @smetl will inform you when it is the case.

**Comment by jvOrsouw** [comment]
For V12 I have a new issue: how to handle changes in tax percentage. We are facing an increase of one of our taxes, and the change will probably come with a date, but also with certain restrictions, f.e. that 'delivery date' instead of 'invoice date' determines what tax percentage has to be used. We need to make new tax codes, we don't need new tags, and we need a way to implement the new taxes to the products and invoices in a correct way. As I understand, there was a module created for Odoo 6 by Holger Brunn with the previous tax change, maybe that is a starting point.

**Comment by lap-odoo** [comment]
@jvOrsouw When there is a change planned for the taxes, you can inform us so that we update the localization accordingly. We will create a new tax with the new percentage (but with the existing tag). We will update the tax report too, if necessary. But we don't plan to add validity dates on the taxes for now. These changes don't happen often, we consider that it is acceptable to manage it manually and we prefer to focus on tasks bringing more added value to our users.

**Comment by smetl** [comment]
Hello,

If you want to test the CBS report: 
http://279881-master-l10n-nl-2-las-1b4748.runbot13.odoo.com/web?reload#view_type=kanban&model=account.journal&menu_id=220&action=386

It's available in _Accounting -> Reporting -> CBS_.

Don't hesitate to report anything seems suspect or wrong.

**Comment by jvOrsouw** [comment]
@smetl odoo passwords of admin and demo user have been altered, can you reset them or provide a new test environment?

**Comment by smetl** [comment]
@jvOrsouw I'm rebuilding the branch.

**Comment by smetl** [comment]
new db for CBS report testing: http://280427-master-l10n-nl-2-las-1b4748.runbot9.odoo.com/web/login

**Comment by ploegvde** [comment]
Hi, to do a good test, can you add information from where you are getting the values for the report. Otherwise it will me a real puzzle to test this.

**Comment by ploegvde** [comment]
Is it possible that someone from Odoo answers? @smetl @lap-odoo

**Comment by smetl** [comment]
Hello, sorry for the delay.

My work follows exactly the requirements provided in https://www.cbs.nl/en-gb/deelnemers%20enquetes/overzicht/bedrijven/onderzoek/lopend/international-trade-in-goods/idep-code-lists

The computations are mainly inspired from an OCA module:
https://github.com/OCA/l10n-netherlands/tree/10.0/l10n_nl_cbs_export

If you want to take a look to the code, see https://github.com/odoo/odoo/compare/master...odoo-dev:master-l10n_nl-2-las

I hope it helps. Don't hesitate if you have more specific questions.

**Comment by ploegvde** [comment]
Sorry, can't login to test build. Can you create new test build or share login credentials?

**Comment by jvOrsouw** [comment]
@smetl we can't login to [http://280427-master-l10n-nl-2-las-1b4748.runbot9.odoo.com/web/login] because the admin pw has changed. Can you reset the pw or give us a new test env please?

**Comment by smetl** [comment]
Hello,

You can test on:
http://287080-master-l10n-nl-2-las-1b4748.runbot11.odoo.com/?db=287080-master-l10n-nl-2-las-1b4748-all

same login/password: admin/admin

Thanks

**Comment by jvOrsouw** [comment]
Tested the module and I have the following suggestions on file addons/l10n_nl/models/l10n_nl_cbs_wizard.py

line 44: registration number: this should be the registration number of the company with CBS, so please create a field at company level (f.e. on an extra sheet 'CBS' on view res.company.form), and get the registration number from there
line 45: version number: version number of Odoo, similar to the information in xaf_audit_file
line 69: mode of transport: it is now set to '3', but can have a value from 1 to 8. This is different per invoice. My suggestion is expand the delivery method and link a 'mode of transport' to a delivery method, and that way get the 'mode of transport' to the report.
line 70: container: is now set to '0', but this can also be the value '1'. My suggestion is to also add this to the delivery method, similar to 'mode of transport'
line 72: statistical procedure: is now set to '00', but can also have a different value per invoice. My suggestion would be to also add this field to the extra 'CBS' sheet on res.company.form, and to show this value on all invoices and have it editable in all states of the invoice
line 73: transaction: is now set to '1', but can have a different value if it is a refund delivery. My suggestion is the same solution as for line 72
line 83: supplementary unit: is now set to '00000000'. this value is only permitted is there is no special criterion. This special criterion is set on the intrastat code, so that table should be expanded with an extra field. The special criterion is a unit of measure, if neccesary, there has to be a link between the product UoM and the special criterion UoM. 
[https://www.cbs.nl/-/media/cbsvooruwbedrijf/international%20trade%20in%20goods/rlo_2018_en.pdf?la=en-gb](url)

**Comment by smetl** [comment]
Hello,

I have a little problem with the CBS report. The following line use a field that is defined in the `report_intrastat` module but this one has a dependency to `stock`, `sale` and `purchase`. It means it add a lot of extra dependencies for the `l10n_nl` module.

https://github.com/odoo/odoo/compare/master...odoo-dev:master-l10n_nl-2-las#diff-235af30a89dbfc57519e4866147ed013R65

Could you check if this line is really necessary plz?

The `Commodity Code` is described in the official website as 

"The 8-digit commodity code according to the Combined Nomenclature.
This code is identical in all member states of the EU. The Combined
Nomenclature is available on the internet (www.cbs.nl/cn). The Combined
Nomenclature is also issued by Eurostat, the Ministry of Finance and the
firm of Kluwer/Samsom." 

in https://www.cbs.nl/en-gb/deelnemers%20enquetes/overzicht/bedrijven/onderzoek/lopend/international-trade-in-goods/idep-code-lists

Thanks in advance.

**Comment by jvOrsouw** [comment]
@smetl Your link doesn't bring me to a line to be discussed. Could you please clearify your problem?

**Comment by Yenthe666** [comment]
@smetl why don't you inline show the code here? With a copy permalink?

**Comment by Yenthe666** [comment]
Hi @lap-odoo,

It looks like the current ICP report in Odoo is having some issues and it could be improved too.
- The amounts on the ICP report can be rounded and shown with nothing behind the decimal precision. If it is for example 49.49 it should be rounded down and become 49. When it is 49.98 it should become 50.
- When a customer has no amount in the product or in the services column the column should be empty (not even containing 0) as there is no booking for that side on that customer.
- On the official report the country code is shown before the VAT number, not the other way around.
- The official label is "Country code", not "Code".
- The country code should not be added in the VAT number. At this moment it shows something like `BE0477472701` while it should be `0477472701`, without the country code.
- When the balance of the debit and credit sides are both identical (which means balance is zero) then the row should not be shown in the ICP report.
- Ideally the VAT number is formatted in groups of 4, so that `0477472701` becomes `0477 4727 01`, but that is just a neatpick.

@lap-odoo can you let me know how you would like to discuss/implement this further?

**Comment by astirpe** [comment]
Hi all,
any chance to have this PR https://github.com/odoo/odoo/pull/23119 merged?

**Comment by ploegvde** [comment]
@smetl Hi, the whole development of the CBS export is stopped. Can we try to finish it for V12 release?

**Comment by astirpe** [comment]
There is a proposal for the CBS export V11 here: https://github.com/OCA/l10n-netherlands/pull/152

**Comment by ploegvde** [comment]
@smetl What would be the ETA so that we can test?

**Comment by smetl** [comment]
@ploegvde I'm rebasing the branch and adapting the code with the generic intrastat report. It will be ready to test very soon.
Thanks for your patience.

**Comment by smetl** [comment]
@jvOrsouw Ok. I added these changes to my branch.

**Comment by ploegvde** [comment]
@smetl Shit, i did not get a ping, so missed your update. The testversion seems to be broken, can you fix or create new one, then i will test it right away.

**Comment by ploegvde** [comment]
@smetl Hi, can you make a new tets instance??
CC @jvOrsouw

**Comment by jvOrsouw** [comment]
@smetl you mean you are fixing the master and then creating a new testbuild?

**Comment by smetl** [comment]
@jvOrsouw @ploegvde You should be able to test on:
http://366711-master-cbs-nl-las-17ab42.runbot11.odoo.com/?db=366711-master-cbs-nl-las-17ab42-all

Sorry for the delay.

**Comment by ploegvde** [comment]
Hi, i get the same screen as before, see image. Seems broken

![image](https://user-images.githubusercontent.com/7593317/44955787-89ddd700-aeb9-11e8-9ea4-a0a5bee97a76.png)

**Comment by ploegvde** [comment]
@smetl Can you have a quick look. Want to have this feature for V12!

**Comment by smetl** [comment]
New test build: http://368489-master-cbs-nl-las-2380a4.runbot16.odoo.com/web/login

**Comment by jvOrsouw** [comment]
Last time I tested CBS report and profit loss report it was looking good. Please merge these reports to V12.

**Comment by ploegvde** [comment]
@smetl Hi, can you merge this so that we can use it in the upcoming V12. Would be terrible if we mis the release

**Comment by ploegvde** [comment]
@smetl Hi Laurent, can you respond?

**Comment by smetl** [comment]
@ploegvde Don't worry. We are working on that.

**Comment by smetl** [comment]
The CBS has been merged on saas-11.5. Thank you for your testing.

**Comment by CasVissers-360ERP** [comment]
I think this issue can be closed? :)

**Comment by Yenthe666** [comment]
Yeah let's close it, out of date & creator left Odoo already too :)
